### ISSUE TITLE
 System Identification Code Missing for Simulink Model in Tools 

### ISSUE THREAD

**Issue by n-shubham** [ArduPilot/ardupilot#22704]
System Identification Code Missing for Simulink Model in Tools 

Currently, the simulink model seems to be using dummy models for simulation.
I see that there is some code to load preexisting models in `sid_sim_init.m` but no code to generate these models. Can we add these functions? 

Further, would appreciate inputs on how to use the simulation to tune a copter's parameters.

@fbredeme @amilcarlucas

**Comment by amilcarlucas** [comment]
Yes, it uses dummy transferfunctions, because you have not yet provided it wit the required data to perform system identification.

Once you do testfligjhts and provide that data, the sripts will replace the simplified transfer functions with accurate transfer functions from YOUR copter.
You can, should and must add that data.

https://github.com/fbredeme/ardupilot_wiki/blob/pr-system-id-improvements/copter/source/docs/systemid-mode-operation.rst
https://github.com/fbredeme/ardupilot_wiki/blob/pr-system-id-improvements/copter/source/docs/systemid-mode-ctrl-optimization.rst

**Comment by fbredeme** [comment]
@amilcarlucas To be precise, the script does not yet contain our system identification functionality. For now, you need to identify the Copter by yourself @n-shubham

**Comment by hunt0r** [comment]
(3 years later)
This looks like a simple case of a wrong assumption, clarified by `@fbredeme`. Should this issue be closed?

Maybe a small outcome could be adding a emphasis box (aka `.. note::`) with the disclaimer: "The actual parameter identification must be performed by the user. This mode is for obtaining data suitable for that." (Or similar.)

Or maybe modify the header sentence?
> This mode [...] provides a means to characterize the response of a vehicle in the frequency domain.

A rewrite like "This mode enables users to generate data suitable for frequency-domain model identification" might be enough clarity? Wordsmithing welcome!
